### PR TITLE
Fix entity cleanup paging

### PR DESCRIPTION
**Pull request by andystaples**
## Summary

- Re-query the current entity cleanup page after deleting empty entities so shifted rows are not skipped.
- Advance to the next page only when the current page contains no deletable entities.
- Deduplicate orphaned-lock release requests when cleanup revisits a page.
- Add an integration regression test covering 150 empty entities across multiple pages.

## Validation

- `dotnet build DurableTask.SqlServer.sln --no-restore --nologo --verbosity minimal`
- The targeted regression test builds but cannot complete locally because SQL Server rejects dynamically created integration-test logins; the pre-existing `EntityQueueModes.SwitchingToSeparateQueuesPartitionsExistingWork` test fails with the same login error.

### CODE REVIEW

File: src/DurableTask.SqlServer/EntitySqlBackendQueries.cs [hunk 1]
```diff
@@ -105,10 +106,8 @@ public async override Task<CleanEntityStorageResult> CleanEntityStorageAsync(Cle
                     CreatedTimeTo = DateTime.MaxValue,
                     FetchInput = true,
```
Comment (bachuv) [inline src/DurableTask.SqlServer/EntitySqlBackendQueries.cs]: Can `FetchInput` be false here?

Comment (andystaples) [reply src/DurableTask.SqlServer/EntitySqlBackendQueries.cs]: Yes. Cleanup reads entity metadata from custom status, so I changed `FetchInput` to `false` to avoid loading unused input payloads.

**Review by bachuv — APPROVED** [review]
